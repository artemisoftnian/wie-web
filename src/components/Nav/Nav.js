/* eslint-disable max-len */
/* eslint-disable prettier/prettier */
import React, { useState, useRef, useEffect } from 'react'
import Link from 'gatsby-link'
import FluidImage from '../FluidImage'

import './nav.css'

function Nav() {
  const navRef = useRef()
  const [isNavOpen, setIsNavOpen] = useState(false)
  const [opacity, setOpacity] = useState(0.9)

  useEffect(() => {
    document.addEventListener('scroll', handleScroll)

    return () => document.removeEventListener('scroll', handleScroll)
  })

  function handleScroll() {
    const winScroll = document.body.scrollTop || document.documentElement.scrollTop
    if (winScroll === 0) {
      setOpacity(0.9)
    }
    if (winScroll > 0) {
      setOpacity(1)
    }
  }   
 
  return (
    <nav
      ref={navRef}
      id="fixedTopNav"
      className="navbar navbar-fixed-top main-navigation navbar-solid"
      itemScope=""
      itemType="https://schema.org/SiteNavigationElement"
      role="navigation"
      style={{ opacity: opacity }}
    >
      <div className="container">
        <div className="navbar-header">
          <div
            className="navbar-brand"
            style={{ padding: '5px 15px 5px' }}
            itemScope=""
            itemType="https://schema.org/Organization"
          >
            <span className="sr-only">WieCare</span>
            <Link to="/">
              <FluidImage  
                fileName="wie_logo_white.png"                
                alt="WieCare Logo"
                className="logo_holder"
              />
              <div className="homeLink">Work Initiative And Entrepreneurship</div>
            </Link>
          </div>
          <button
            type="button"
            id="hamburger-1"
            onClick={() => setIsNavOpen(!isNavOpen)}
            className={`navbar-toggle collapsed hamburger ${isNavOpen ? 'is-active' : ''}`}
            aria-expanded={isNavOpen ? 'true' : 'false'}
          >
            <span className="line"></span>
            <span className="line"></span>
            <span className="line"></span>
            <span className="sr-only">WieCare</span>
          </button>
        </div>
        <div
          className={`navbar-collapse collapse ${isNavOpen ? 'in' : ''}`}
          id="main-nav-collapse"
          aria-expanded={isNavOpen ? 'true' : 'false'}
        >
          <ul className="nav navbar-nav navbar-right" role="menu">
            <li role="menuitem" className="donate">
              <Link to="/">
                <span>Home</span>
              </Link>
            </li>
            <li role="menuitem" className="donate">
              <Link to="/about">
                <span>About Us</span>
              </Link>
            </li>
            <li role="menuitem" className="donate">
              <Link to="/donate">
                <span>Get Involved</span>
              </Link>
            </li>
            <li role="menuitem" className="donate">
              <Link to="/programs">
                <span>Programs</span>
              </Link>
            </li> 
            <li role="menuitem" className="donate">
              <Link to="/projects">
                <span>Projects</span>
              </Link>
            </li>
            <li role="menuitem" className="donate">
              <Link to="/events">
                <span>Events</span>
              </Link>
            </li>
          </ul>

        </div>
      </div>
    </nav>
  )
}

export default Nav
