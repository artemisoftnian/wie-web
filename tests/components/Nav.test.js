import React from 'react'
import Nav from '../../src/components/Nav'
import { render } from '@testing-library/react'

describe('<Nav />', () => {
  test('should render correctly', () => {
    const { container } = render(<Nav />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
