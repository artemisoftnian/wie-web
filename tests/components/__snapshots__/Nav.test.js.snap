// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render correctly 1`] = `
<nav
  class="navbar navbar-fixed-top main-navigation navbar-solid"
  id="fixedTopNav"
  itemtype="https://schema.org/SiteNavigationElement"
  role="navigation"
  style="opacity: 0.9;"
>
  <div
    class="container"
  >
    <div
      class="navbar-header"
    >
      <div
        class="navbar-brand"
        itemtype="https://schema.org/Organization"
        style="padding: 5px 15px 5px;"
      >
        <span
          class="sr-only"
        >
          WieCare
        </span>
        <a
          aria-current="page"
          class=""
          href="/"
        >
          static-query
          <div
            class="homeLink"
          >
            Work Initiative And Entrepreneurship
          </div>
        </a>
      </div>
      <button
        aria-expanded="false"
        class="navbar-toggle collapsed hamburger "
        id="hamburger-1"
        type="button"
      >
        <span
          class="line"
        />
        <span
          class="line"
        />
        <span
          class="line"
        />
        <span
          class="sr-only"
        >
          WieCare
        </span>
      </button>
    </div>
    <div
      aria-expanded="false"
      class="navbar-collapse collapse "
      id="main-nav-collapse"
    >
      <ul
        class="nav navbar-nav navbar-right"
        role="menu"
      >
        <li
          class="donate"
          role="menuitem"
        >
          <a
            aria-current="page"
            class=""
            href="/"
          >
            <span>
              Home
            </span>
          </a>
        </li>
        <li
          class="donate"
          role="menuitem"
        >
          <a
            href="/about"
          >
            <span>
              About Us
            </span>
          </a>
        </li>
        <li
          class="donate"
          role="menuitem"
        >
          <a
            href="/about"
          >
            <span>
              Get Involved
            </span>
          </a>
        </li>
        <li
          class="donate"
          role="menuitem"
        >
          <a
            href="/about"
          >
            <span>
              Programs
            </span>
          </a>
        </li>
        <li
          class="donate"
          role="menuitem"
        >
          <a
            href="/about"
          >
            <span>
              Proyects
            </span>
          </a>
        </li>
        <li
          class="donate"
          role="menuitem"
        >
          <a
            href="/about"
          >
            <span>
              Events
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
